source/common/redis/command_splitter_impl.cc

```diff
@@ -145,7 +167,8 @@ void MGETRequest::onChildResponse(RespValuePtr&& value, uint32_t index) {
 SplitRequestPtr MSETRequest::create(ConnPool::Instance& conn_pool,
                                     const RespValue& incoming_request, SplitCallbacks& callbacks) {
   if ((incoming_request.asArray().size() - 1) % 2 != 0) {
-    callbacks.onResponse(Utility::makeError("wrong number of arguments for command"));
+    callbacks.onResponse(Utility::makeError(fmt::format(
```

Comment:
nit: Can you share this error creation with the other one and put in base class? It's same message.